climada_petals/hazard/copernicus_interface/time_utils.py

```diff
@@ -0,0 +1,156 @@
+"""
+This file is part of CLIMADA.
+
+Copyright (C) 2017 ETH Zurich, CLIMADA contributors listed in AUTHORS.
+
+CLIMADA is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free
+Software Foundation, version 3.
+
+CLIMADA is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along
+with CLIMADA. If not, see <https://www.gnu.org/licenses/>.
+
+---
+Time utility functions for seasonal forecast pipelines in CLIMADA.
+Provides helpers to convert month names to numbers and calculate lead times
+based on forecast configuration.
+"""
+
+import calendar
+from datetime import date
+import numpy as np
+
+##########  Utility Functions  ##########
+
+
+def month_name_to_number(month):
+    """
+    Convert a month name or number to its corresponding integer value.
+
+    Accepts either an integer (1-12), full month name (e.g., 'March'),
+    or abbreviated month name (e.g., 'Mar') and returns the corresponding
+    month number (1-12).
+
+    Parameters
+    ----------
+    month : int or str
+        Month as an integer (1-12) or as a string (full or abbreviated month name).
+
+    Returns
+    -------
+    int
+        Month as an integer in the range 1-12.
+
+    Raises
+    ------
+    ValueError
+        If the input month is invalid, empty, or outside the valid range.
+    """
+    if isinstance(month, int):  # Already a number
+        if 1 <= month <= 12:
+            return month
+        else:
+            raise ValueError("Month number must be between 1 and 12.")
+    if isinstance(month, str):
+        if not month.strip():
+            raise ValueError("Month cannot be empty.")  # e.g. "" or "   "
+        month = month.capitalize()  # Ensure consistent capitalization
+        if month in calendar.month_name:
+            return list(calendar.month_name).index(month)
+        elif month in calendar.month_abbr:
+            return list(calendar.month_abbr).index(month)
+    raise ValueError(f"Invalid month input: {month}")
```

Comment:
Please add a specific ValueError in case of `month` is a `str`, but wrong, under the `if isinstance(month, str)` block. E.g.
```
elif month in calendar.month_abbr:
     return list(calendar.month_abbr).index(month)
else:
     raise ValueError(f"Invalid string format for month input: {month}")
```

and maybe add something like: `... of type: {type(month)}"` in your current ValueError